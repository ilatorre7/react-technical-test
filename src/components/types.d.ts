export interface IItem {
  id: number,
  title: string,
  image: string,
  description: string,
  detail: {
    brand: string,
    price: string,
    year: number,
    colorway: string
  }
}

export interface IOption { 
  label: string, 
  value: string 
}