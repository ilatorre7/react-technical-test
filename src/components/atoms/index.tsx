import { InputSelect } from "./InputSelect";
import { Button } from "./Button";

export { InputSelect, Button }