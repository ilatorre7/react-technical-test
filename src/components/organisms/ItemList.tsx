import "../../styles/ItemList.css"
import { IItem } from "../types"
import { ItemCard } from "../molecules/ItemCard"

export const ItemList = ({ list, isLoading }: { list: IItem[], isLoading: boolean }) => {

  return (
    <div className='list-container'>
      {
        list.map((item) => (
          <ItemCard key={`item-${item.id}`} {...item} />
        ))
      }
      {isLoading && <p>Loading...</p>}
    </div>
  )
}