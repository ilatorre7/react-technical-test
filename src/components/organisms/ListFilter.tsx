import '../../styles/Aside.css'
import { IOption } from "../types"
import { InputSelect } from "../atoms/InputSelect"

export const ListFilter = ({ options, onFilter } : { options: IOption[], onFilter : (e: string) => void } ) => {

  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    e.preventDefault()
    onFilter(e.target.value)
  }

  return (
    <aside className='aside'>
      <h3>Filtrar por:</h3>
      <InputSelect placeholder='Selecciona una marca para filtrar' options={options} onChange={handleChange} />
    </aside>
  )
}