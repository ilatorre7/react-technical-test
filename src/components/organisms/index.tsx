import { ItemList } from "./ItemList";
import { ListFilter } from "./ListFilter";

export { ItemList, ListFilter }