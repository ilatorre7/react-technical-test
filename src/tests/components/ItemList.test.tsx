import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react';
import { ItemList } from '../../components/organisms';

describe('<ItemList />', () => {
  const expectedCards = [{
    id: 1,
    title: "Jordan 1 Low OG",
    image: "https://nikearprod.vtexassets.com/arquivos/ids/880101-800-800?width=800&height=800&aspect=true",
    description: "Comienza el Año del Dragón con estilo. Este elegante AJ1 tiene superposiciones escamosas en verde petróleo inspiradas en la única criatura mitológica del zodíaco chino.",
    detail: {
      price: "299999",
      year: 2023,
      brand: "Jordan",
      colorway: "Verde petróleo/Metallic Gold"
    }
  },
  {
    id: 2,
    title: "Jordan 1 Mid SE",
    image: "https://nikearprod.vtexassets.com/arquivos/ids/811855-1200-1200?width=1200&height=1200&aspect=true",
    description: "Una versión moderna de un clásico, el Jordan 1 Mid SE cuenta con materiales premium y detalles únicos.",
    detail: {
      price: "279999",
      year: 2022,
      brand: "Jordan",
      colorway: "Blanco/Negro/Rojo"
    }
  }]

  it('Should render a list of cards with the correct values', () => {
    const { container } = render(<ItemList list={expectedCards} isLoading={false} />)
    expect(container).toMatchSnapshot()
  })
})