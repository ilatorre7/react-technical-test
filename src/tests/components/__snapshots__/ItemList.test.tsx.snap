// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ItemList /> > Should render a list of cards with the correct values 1`] = `
<div>
  <div
    class="list-container"
  >
    <article
      class="card-container"
    >
      <div
        class="card-content"
      >
        <img
          alt="Jordan 1 Low OG"
          class="card-image"
          src="https://nikearprod.vtexassets.com/arquivos/ids/880101-800-800?width=800&height=800&aspect=true"
        />
        <div
          class="card-info"
        >
          <header
            class="card-header"
          >
            <h3
              class="card-title"
            >
              Jordan 1 Low OG
            </h3>
            <h4>
              2023
            </h4>
          </header>
          <p>
            Comienza el Año del Dragón con estilo. Este elegante AJ1 tiene superposiciones escamosas en verde petróleo inspiradas en la única criatura mitológica del zodíaco chino.
          </p>
          <h4
            class="card-price"
          >
            $299.999
          </h4>
          <button
            class="button-container"
          >
            Detalles
          </button>
        </div>
      </div>
    </article>
    <article
      class="card-container"
    >
      <div
        class="card-content"
      >
        <img
          alt="Jordan 1 Mid SE"
          class="card-image"
          src="https://nikearprod.vtexassets.com/arquivos/ids/811855-1200-1200?width=1200&height=1200&aspect=true"
        />
        <div
          class="card-info"
        >
          <header
            class="card-header"
          >
            <h3
              class="card-title"
            >
              Jordan 1 Mid SE
            </h3>
            <h4>
              2022
            </h4>
          </header>
          <p>
            Una versión moderna de un clásico, el Jordan 1 Mid SE cuenta con materiales premium y detalles únicos.
          </p>
          <h4
            class="card-price"
          >
            $279.999
          </h4>
          <button
            class="button-container"
          >
            Detalles
          </button>
        </div>
      </div>
    </article>
  </div>
</div>
`;
